tools/wptrunner/wptrunner/tests/test_testloader.py

```diff
@@ -135,6 +168,110 @@ def test_filter_unicode():
         TestFilter(manifest_path=f.name, test_manifests=tests)
 
 
+def test_filter_directory(manifest1):
+    with tempfile.TemporaryDirectory() as tmp:
+        include = os.path.normpath(os.path.join(tmp, "a", "b"))
+        os.makedirs(include)
+        test_filter = TestFilter({manifest1: {"tests_path": tmp}}, include=[include])
+        items = sorted(test_filter(manifest1), key=lambda item: item[1])
+    assert len(items) == 2
+    test_type, test_path, tests = items[0]
+    assert test_type == "testharness"
+    assert test_path == os.path.join("a", "b", "bar.html")
+    assert {test.id for test in tests} == {"/a/b/bar.html"}
+    test_type, test_path, tests = items[1]
+    assert test_type == "testharness"
+    assert test_path == os.path.join("a", "b", "c.html")
+    assert {test.id for test in tests} == {"/a/b/c.html"}
+
+
+def test_filter_variants_for_scope(manifest1):
+    with tempfile.TemporaryDirectory() as tmp:
+        test_filter = TestFilter({manifest1: {"tests_path": tmp}},
+                                 include=["/a/foo.https.any.html"])
+        items = list(test_filter(manifest1))
+    assert len(items) == 1
+    test_type, test_path, tests = items[0]
+    assert test_type == "testharness"
+    assert test_path == os.path.join("a", "foo.https.any.js")
+    assert {test.id for test in tests} == {
+        "/a/foo.https.any.html?b",
+        "/a/foo.https.any.html?c",
+    }
+
+
+def test_filter_multiglobal_file(manifest1):
+    with tempfile.TemporaryDirectory() as tmp:
+        include = os.path.normpath(os.path.join(tmp, "a", "foo.https.any.js"))
+        os.makedirs(os.path.dirname(include))
+        with open(include, "w"):
+            pass
+        test_filter = TestFilter({manifest1: {"tests_path": tmp}}, include=[include])
+        items = list(test_filter(manifest1))
+    assert len(items) == 1
+    test_type, test_path, tests = items[0]
+    assert test_type == "testharness"
+    assert test_path == os.path.join("a", "foo.https.any.js")
+    assert {test.id for test in tests} == {
+        "/a/foo.https.any.html?b",
+        "/a/foo.https.any.html?c",
+        "/a/foo.https.any.worker.html?b",
+        "/a/foo.https.any.worker.html?c",
+    }
+
+
+def test_filter_glob(manifest1):
+    with tempfile.TemporaryDirectory() as tmp:
+        include = os.path.normpath(os.path.join(tmp, "a", "b"))
+        os.makedirs(include)
+        for filename in ["bar.html", "c.html"]:
```

Comment:
Can we have a common setup function for all tests? Not sure we need to create the files here, but not in `test_filter_directory`.